### ISSUE TITLE
Retinanet error, probably cause by the loss function. 

### ISSUE THREAD

**Issue by icoicqico** [Stick-To/Object-Detection-API-Tensorflow#36]
Caused by op 'inference/while/ArgMax_1', defined at:
  File "testretinanet.py", line 76, in <module>
    retinanet = net.RetinaNet(config, trainset_provider)
  File "F:\tensorflow-network\RetinaNet.py", line 73, in __init__
    self._build_detection_graph()
  File "F:\tensorflow-network\RetinaNet.py", line 210, in _build_detection_graph
    state = tf.while_loop(cond, body, init_state)
  File "e:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\ops\control_flow_ops.py", line 3556, in while_loop
    return_same_structure)
  File "e:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\ops\control_flow_ops.py", line 3087, in BuildLoop
    pred, body, original_loop_vars, loop_vars, shape_invariants)
  File "e:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\ops\control_flow_ops.py", line 3022, in _BuildLoop
    body_result = body(*packed_vars_for_body)
  File "F:\tensorflow-network\RetinaNet.py", line 205, in <lambda>
    tf.squeeze(tf.gather(self.ground_truth, tf.cast(i, tf.int32))),
  File "F:\tensorflow-network\RetinaNet.py", line 418, in _compute_one_image_loss
    rgindex = tf.argmax(other_agiou_rate,axis=1)
  File "e:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\util\deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "e:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\ops\math_ops.py", line 86, in argmax
    return argmax_v2(input, axis, output_type, name)
  File "e:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\ops\math_ops.py", line 115, in argmax_v2
    return gen_math_ops.arg_max(input, axis, name=name, output_type=output_type)
  File "e:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\ops\gen_math_ops.py", line 916, in arg_max
    name=name)
  File "e:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\framework\op_def_library.py", line 788, in _apply_op_helper
    op_def=op_def)
  File "e:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\util\deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "e:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 3300, in create_op
    op_def=op_def)
  File "e:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 1801, in __init__
    self._traceback = tf_stack.extract_stack()

InvalidArgumentError (see above for traceback): Reduction axis 1 is empty in shape [32526,0]
         [[node inference/while/ArgMax_1 (defined at F:\tensorflow-network\RetinaNet.py:418) ]]
         [[node inference/add_21 (defined at F:\tensorflow-network\RetinaNet.py:217) ]]

**Comment by Stick-To** [comment]
don't use  zoom size  in image augmentor config

**Comment by icoicqico** [comment]
I didn't use zoom size, and if I change the rgindex = tf.argmax(other_agiou_rate,axis=1) to rgindex = tf.argmax(other_agiou_rate), the program can run but the loss= Nan.

**Comment by Stick-To** [comment]
try use image augmentor
only pad truth to
and data format

**Comment by Stick-To** [comment]
and  output  shape

**Comment by icoicqico** [comment]
I have commented other settings still got the same error.

**Comment by icoicqico** [comment]
image_augmentor_config = {
    'data_format': 'channels_last',
    'output_shape': [416, 416],
    #'zoom_size': [520, 520],
    #'crop_method': 'random',
    #'flip_prob': [0., 0.5],
    #'fill_mode': 'BILINEAR',
    #'keep_aspect_ratios': True,
    #'constant_values': 0.,
    #'color_jitter_prob': 0.5,
    #'rotate': [0.5, -5., -5.],
    'pad_truth_to': 60,
}

**Comment by xiao-an-qi** [comment]
@icoicqico  hey, have you solve the problem?

**Comment by Stick-To** [comment]
@Andy457  image_augmemtor have some bugs, it is fixed now

**Comment by xiao-an-qi** [comment]
@Stick-To only change batch_size , buffer_size(both lower) and classes, it throws this error:Incompatible shapes: [4,38,38,256] vs. [4,32,32,256], need any fixed size?

**Comment by xiao-an-qi** [comment]
@Stick-To similiar error for SSD300/SSD512
